### PR TITLE
elia: init at 1.9.0

### PR DESCRIPTION
**Pull request by Daru-san**
## Description of changes

Add elia, a TUI for interacting with large language models like ChatGPT
Homepage: https://github.com/darrenburns/elia

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/el/elia/package.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "elia";
+  version = "1.7.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "darrenburns";
+    repo = "elia";
+    rev = version;
```
Comment (fabaff) [inline pkgs/by-name/el/elia/package.nix:15]: Helps to handle branch/tag confusion.

```suggestion
    rev = "refs/tags/${version}";
```

File: pkgs/by-name/el/elia/package.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    hash = "sha256-hQe/0M58JfSC7t0hPVsN0rNeHQh1LUccj/CxR+1RbEw=";
+  };
+
+  nativeBuildInputs = [ python3.pkgs.hatchling ];
```
Comment (fabaff) [inline pkgs/by-name/el/elia/package.nix:19]:
```suggestion
  build-system = [ python3.pkgs.hatchling ];
```

File: pkgs/by-name/el/elia/package.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  propagatedBuildInputs = with python3.pkgs; [
```
Comment (fabaff) [inline pkgs/by-name/el/elia/package.nix:21]:
```suggestion
  dependencies = with python3.pkgs; [
```

File: pkgs/by-name/el/elia/package.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    sqlmodel
+    textual
+    xdg-base-dirs
+  ];
+
+  pythonImportsCheck = [ "elia_chat" ];
+
+  meta = with lib; {
+    description = "A snappy, keyboard-centric terminal user interface for interacting with large language models";
+    homepage = "https://github.com/darrenburns/elia";
```
Comment (fabaff) [inline pkgs/by-name/el/elia/package.nix:39]:
```suggestion
    homepage = "https://github.com/darrenburns/elia";
    changelog = "https://github.com/darrenburns/elia/releases/tag/${version}";
```

File: pkgs/by-name/el/elia/package.nix (hunk 1 above)
Comment (Daru-san) [reply pkgs/by-name/el/elia/package.nix]: Thanks, I've made all of the relevant changes but it seems the package will not be able to build until this is merged [darrenburns/elia#47](https://github.com/darrenburns/elia/pull/47), due to the version of the textual dependency

**Comment by Daru-san** [comment]
I think it might also be helpful to temporarily pin the textual dependency to the version supported upstream until it has been updated
 
```nix
...
  dependencies = [("other deps")]
    ++ [
    # Pin until upstream updates this dependency
      (python3.pkgs.textual.overrideAttrs (finalAttrs: {
        version = "0.62.0";
        src = fetchFromGitHub {
          owner = "Textualize";
          repo = "textual";
          rev = "v${finalAttrs.version}";
          hash = "sha256-3zMA2gZMvXTDOPVyMf/i4onumkHL85om1rvfnBN4VEE=";
        };
      }))
    ];
...
```

**Comment by FliegendeWurst** [comment]
@ofborg build elia

File: pkgs/by-name/el/elia/package.nix [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+          rev = "v${finalAttrs.version}";
+          hash = "sha256-3zMA2gZMvXTDOPVyMf/i4onumkHL85om1rvfnBN4VEE=";
+        };
+      }))
+    ];
+
+  pythonImportsCheck = [ "elia_chat" ];
+
+  meta = with lib; {
+    description = "A snappy, keyboard-centric terminal user interface for interacting with large language models";
```
Comment (FliegendeWurst) [inline pkgs/by-name/el/elia/package.nix:50]:
```suggestion
    description = "Snappy, keyboard-centric terminal user interface for interacting with large language models";
```

Comment (Daru-san) [reply pkgs/by-name/el/elia/package.nix]: Thanks for pointing that out, missed that

**Comment by Daru-san** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 317782`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>elia</li>
    <li>elia.dist</li>
  </ul>
</details>

File: pkgs/by-name/el/elia/package.nix [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    owner = "darrenburns";
+    repo = "elia";
+    rev = "refs/tags/${version}";
+    hash = "sha256-tf5FU1k140g84/BxruFXQS2h3kL1hSCvDv9kljPI410=";
+  };
+
+  postPatch = ''
+    substituteInPlace ./pyproject.toml \
+      --replace-fail '"textual[syntax]==0.79.1",' '"textual[syntax]>=0.79.1",'
+  '';
```
Comment (FliegendeWurst) [inline pkgs/by-name/el/elia/package.nix:22]: have you tried `pythonRelaxDeps`?

**Comment by Daru-san** [comment]
Sorry about that, rebased incorrectly

**Pull request closed without merging**